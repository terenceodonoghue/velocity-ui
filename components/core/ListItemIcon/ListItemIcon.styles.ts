import { css, FunctionInterpolation, Theme } from '@emotion/react';

const listItemIcon: FunctionInterpolation<Theme> = ({ palette }) => css`
  flex-shrink: 0;
  height: 24px;
  width: 24px;
  color: ${palette.neutral[400]};
`;

export default listItemIcon;
