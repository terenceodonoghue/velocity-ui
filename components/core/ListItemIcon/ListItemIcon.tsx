import React, { FunctionComponent } from 'react';
import listItemIcon from './ListItemIcon.styles';
import type { ListItemIconProps } from './ListItemIcon.types';

const ListItemIcon: FunctionComponent<ListItemIconProps> = ({
  icon: Icon,
  ...props
}) => <Icon css={listItemIcon} data-testid="listitemicon" {...props} />;

export default ListItemIcon;
