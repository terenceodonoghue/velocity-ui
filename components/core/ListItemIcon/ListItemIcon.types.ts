import { ComponentType, SVGProps } from 'react';

export interface ListItemIconProps extends SVGProps<SVGSVGElement> {
  icon: ComponentType<SVGProps<SVGSVGElement>>;
}
