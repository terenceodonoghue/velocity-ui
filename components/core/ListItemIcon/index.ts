export { default } from './ListItemIcon';
export type { ListItemIconProps } from './ListItemIcon.types';
