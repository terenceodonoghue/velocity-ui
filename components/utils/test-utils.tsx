import React, { FunctionComponent, ReactElement } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import ThemeProvider from '~/components/core/ThemeProvider';

const Providers: FunctionComponent = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult => render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

export { customRender as render };
