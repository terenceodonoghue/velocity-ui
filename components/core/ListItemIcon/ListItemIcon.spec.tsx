import React from 'react';
import { render } from 'test-utils';
import ListItemIcon from './ListItemIcon';

describe('ListItemIcon', () => {
  it('renders icon', () => {
    const { container } = render(<ListItemIcon icon={() => <svg />} />);

    expect(container).toMatchSnapshot();
  });
});
