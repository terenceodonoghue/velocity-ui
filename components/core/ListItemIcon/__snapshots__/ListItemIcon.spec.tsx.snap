// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemIcon renders icon 1`] = `
<div>
  <svg />
</div>
`;
